### PR TITLE
Basic Entity Typing and Attributes / Cleanup

### PR DESCRIPTION
**Pull request by nkane**

### CODE REVIEW

File: game/src/entity.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/*
+ *  entity.h
+ *
+ */
+
+typedef enum _entityType
+{
+	NOTYPE      = 0x00,
+	PLAYER      = 0x01,
+    WEAPON      = 0x03,
```
Comment (mikhailswift) [inline game/src/entity.h]: WTF NICK?!

File: game/src/entity.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    EntitySubType subType;
+    EntityAttribute attributes;
+} EntityProp;
+
+typedef struct _entity
+{
+    Vector2 position;
+    Vector2 velocity;
+    float rotation;
+	float maxVelocity;
```
Comment (mikhailswift) [inline game/src/entity.h]: AHHHHH

**Review by mikhailswift — APPROVED** [review]

File: game/src/entity.h (hunk 2 above)
Comment (nkane) [reply game/src/entity.h]: FML!

File: game/src/entity.h (hunk 1 above)
Comment (nkane) [reply game/src/entity.h]: I am pretty sure vim hates me :(

Comment (nkane) [reply game/src/entity.h]: image
WTF?!

**Pull request merged**